Terraform crashing when Modifying Nova instances

openstack_compute_instance_v2.vm_1: Modifying...
  network.0.access_network: "" => "0"
  network.0.port:           "" => "a15e43ca-4023-471c-8be5-37b67bbb6132"
Error applying plan:

1 error(s) occurred:

* openstack_compute_instance_v2.vm_1: unexpected EOF

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
panic: interface conversion: interface {} is nil, not map[string]interface {}
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: 
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: goroutine 114 [running]:
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: panic(0xb2b3a0, 0xc820420cc0)
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: 	/opt/go/src/runtime/panic.go:464 +0x3e6
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: github.com/hashicorp/terraform/builtin/providers/openstack.resourceComputeInstanceV2Update(0xc820307f20, 0xb94f80, 0xc8201e0d80, 0x0, 0x0)
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/openstack/resource_openstack_compute_instance_v2.go:652 +0x5bf8
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc820371c80, 0xc820418600, 0xc82041b1e0, 0xb94f80, 0xc8201e0d80, 0xc820418701, 0x0, 0x0)
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:151 +0x3ae
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc82038c240, 0xc820420480, 0xc820418600, 0xc82041b1e0, 0x1, 0x0, 0x0)
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:162 +0x1ed
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: github.com/hashicorp/terraform/rpc.(*ResourceProviderServer).Apply(0xc820373ac0, 0xc820422600, 0xc82041b220, 0x0, 0x0)
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: 	/opt/gopath/src/github.com/hashicorp/terraform/rpc/resource_provider.go:323 +0x76
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: reflect.Value.call(0xabdb80, 0xbb6198, 0x13, 0xc05870, 0x4, 0xc8202e3ed8, 0x3, 0x3, 0x0, 0x0, ...)
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: 	/opt/go/src/reflect/value.go:435 +0x120d
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: reflect.Value.Call(0xabdb80, 0xbb6198, 0x13, 0xc8202e3ed8, 0x3, 0x3, 0x0, 0x0, 0x0)
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: 	/opt/go/src/reflect/value.go:303 +0xb1
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: net/rpc.(*service).call(0xc8203902c0, 0xc820390280, 0xc82037ad50, 0xc82035c200, 0xc820373ae0, 0x9b75a0, 0xc820422600, 0x16, 0x9b7600, 0xc82041b220, ...)
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: 	/opt/go/src/net/rpc/server.go:383 +0x1c2
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: created by net/rpc.(*Server).ServeCodec
2016/02/26 15:43:14 [DEBUG] terraform-provider-openstack: 	/opt/go/src/net/rpc/server.go:477 +0x49d
2016/02/26 15:43:14 [DEBUG] root: eval: *terraform.EvalWriteState
2016/02/26 15:43:14 [DEBUG] /home/avalz/terraform/terraform-provider-openstack: plugin process exited
2016/02/26 15:43:14 [DEBUG] root: eval: *terraform.EvalApplyProvisioners
2016/02/26 15:43:14 [DEBUG] root: eval: *terraform.EvalIf
2016/02/26 15:43:14 [DEBUG] root: eval: *terraform.EvalWriteDiff
2016/02/26 15:43:14 [DEBUG] root: eval: *terraform.EvalIf
2016/02/26 15:43:14 [DEBUG] root: eval: *terraform.EvalWriteState
2016/02/26 15:43:14 [DEBUG] root: eval: *terraform.EvalApplyPost
2016/02/26 15:43:14 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* openstack_compute_instance_v2.vm_1: unexpected EOF
2016/02/26 15:43:14 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* openstack_compute_instance_v2.vm_1: unexpected EOF
2016/02/26 15:43:14 [ERROR] root: eval: *terraform.EvalOpFilter, err: 1 error(s) occurred:

* openstack_compute_instance_v2.vm_1: unexpected EOF
2016/02/26 15:43:14 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* openstack_compute_instance_v2.vm_1: unexpected EOF
2016/02/26 15:43:14 [TRACE] [walkApply] Exiting eval tree: openstack_compute_instance_v2.vm_1
2016/02/26 15:43:14 [DEBUG] vertex provider.openstack (close), got dep: openstack_compute_instance_v2.vm_1
2016/02/26 15:43:14 [DEBUG] waiting for all plugin processes to complete...

--- a/builtin/providers/openstack/resource_openstack_compute_instance_v2.go
+++ b/builtin/providers/openstack/resource_openstack_compute_instance_v2.go
@@ -654,16 +654,23 @@ func resourceComputeInstanceV2Update(d *schema.ResourceData, meta interface{}) e
 		oldNetworkList := oldNetworks.([]interface{})
 		newNetworkList := newNetworks.([]interface{})
 		for i, oldNet := range oldNetworkList {
-			oldNetRaw := oldNet.(map[string]interface{})
-			oldFIP := oldNetRaw["floating_ip"].(string)
-			oldFixedIP := oldNetRaw["fixed_ip_v4"].(string)
+			var oldFIP, newFIP string
+			var oldFixedIP, newFixedIP string
 
-			newNetRaw := newNetworkList[i].(map[string]interface{})
-			newFIP := newNetRaw["floating_ip"].(string)
-			newFixedIP := newNetRaw["fixed_ip_v4"].(string)
+			if oldNetRaw, ok := oldNet.(map[string]interface{}); ok {
+				oldFIP = oldNetRaw["floating_ip"].(string)
+				oldFixedIP = oldNetRaw["fixed_ip_v4"].(string)
+			}
+
+			if len(newNetworkList) > i {
+				if newNetRaw, ok := newNetworkList[i].(map[string]interface{}); ok {
+					newFIP = newNetRaw["floating_ip"].(string)
+					newFixedIP = newNetRaw["fixed_ip_v4"].(string)
+				}
+			}
 
 			// Only changes to the floating IP are supported
-			if oldFIP != newFIP {
+			if oldFIP != "" && newFIP != "" && oldFIP != newFIP {
 				log.Printf("[DEBUG] Attempting to disassociate %s from %s", oldFIP, d.Id())
 				if err := disassociateFloatingIPFromInstance(computeClient, oldFIP, d.Id(), oldFixedIP); err != nil {
 					return fmt.Errorf("Error disassociating Floating IP during update: %s", err)

